firmware/arduino/chessboard/uart.ino

```diff
@@ -1,84 +1,92 @@
+        // Check if the buffer is full, process input
         if (byteNum == 5)
         {
-            //Reset buffer position
+            // Reset buffer position
             byteNum = -1;
 
-            //Process input
-            //Returns true for valid input and false for invalid input. Calls movement function
+            // Process input
+            // Returns true for valid input and false for invalid input. Calls movement function
             parseMessageFromPi(buffer);
         }
     }
 }
 
 bool parseMessageFromPi(char * message)
 {
-    //Move types 0 and 3 are valid for the same range of values
+    // Move types 0 and 3 are valid for the same range of values
     if (message[0] == 0 || message[0] == 3)
     {
-        //Check that the inputs are in a valid range
-        if (message[1] < 'a' || message[1] > 'h' || message[2] < '1' ||  message[2] > '8' || message[3] < 'a' || message[3] > 'h' || message[4] < '1' ||  message[4] > '8') 
+        // Check that the inputs are in a valid range
+        if (message[1] < 'a' || message[1] > 'h' 
+           || message[2] < '1' ||  message[2] > '8' 
+           || message[3] < 'a' || message[3] > 'h' 
+           || message[4] < '1' ||  message[4] > '8') 
         {
             return false;
         }
 
-        //Move type 0
+        // Move type 0
         if (message[0] == 0)
             moveStraight(message[1] - 'a', message[2] - '1', message[3] - 'a', message[4] - '1');
-        //Move type 3
+        // Move type 3
         else
             moveAlongEdges(message[1] - 'a', message[2] - '1', message[3] - 'a', message[4] - '1');
 
     }
     else if (message[0] == 1)
     {
-        //Check validity for the first 3 bytes
-        if (message[1] < 'a' || message[1] > 'h' || message[2] < '1' ||  message[2] > '8' || ((message[3] != 'b') && message[3] != 'w'))
+        // Check validity for the first 3 bytes
+        if (message[1] < 'a' || message[1] > 'h' 
+           || message[2] < '1' ||  message[2] > '8' 
+           || ((message[3] != 'b') && message[3] != 'w'))
             return false;
         
         //Check for valid piece type
-        if (!(message[4] == 'p' || message[4] == 'r' || message[4] == 'k' || message[4] == 'b' || message[4] == 'q'))
+        if (!(message[4] == 'p' || message[4] == 'r' 
+           || message[4] == 'k' || message[4] == 'b' || message[4] == 'q'))
             return false;
 
-        //Move type 1
+        // Move type 1
         moveToGraveyard(message[1] - 'a', message[2] - '1', message[3], message[4]);        
     }
     else if (message[0] == 2)
     {
-        //Check input validity
-        if (!(message[1] == 'b' || message[1] == 'w') || message[2] != '0' || message[4] != '0' || !(message[3] == '0' || message[3] == '-'))
+        // Check input validity
+        if (!(message[1] == 'b' || message[1] == 'w') 
+           || message[2] != '0' || message[4] != '0' 
+           || !(message[3] == '0' || message[3] == '-'))
```

Comment:
@chasesza why did you do another negation here? Would it make less sense to move that out of the initial negation? It's up to you but I'm suggesting turing !(a  ||  !(b  ||  c)) into !a  &&  (b  ||  c)
Where a is that first set of stuff.

Reply:
I think it's more readable if we check that it's !( the valid options) for each character.

Also, there isn't a double not the way you listed. It's !(a || b) || c || d || !(e || f)

Reply:
Ohhhhh, my bad, I misread that. Then this is all good.

Reply:
It might be easier to read if you do `(!a && !b)` for the two cases where you currently have `!(a || b)` to distinguish it from the other conditions you're checking. That way each condition is separated by an `||`